test/custom-parser.5.test.js

```diff
@@ -1,43 +1,40 @@
 'use strict'
 
-const t = require('tap')
-const test = t.test
+const { test } = require('node:test')
 const sget = require('simple-get').concat
 const Fastify = require('../fastify')
 const jsonParser = require('fast-json-body')
 const { getServerUrl, plainTextParser } = require('./helper')
 
 process.removeAllListeners('warning')
 
-test('cannot remove all content type parsers after binding', t => {
+test('cannot remove all content type parsers after binding', (t, done) => {
   t.plan(2)
 
   const fastify = Fastify()
 
-  t.teardown(fastify.close.bind(fastify))
-
   fastify.listen({ port: 0 }, function (err) {
-    t.error(err)
-
-    t.throws(() => fastify.removeAllContentTypeParsers())
+    t.assert.ifError(err)
+    t.assert.throws(() => fastify.removeAllContentTypeParsers())
+    fastify.close()
+    done()
   })
 })
 
-test('cannot remove content type parsers after binding', t => {
+test('cannot remove content type parsers after binding', (t, done) => {
   t.plan(2)
 
   const fastify = Fastify()
 
-  t.teardown(fastify.close.bind(fastify))
-
   fastify.listen({ port: 0 }, function (err) {
-    t.error(err)
-
-    t.throws(() => fastify.removeContentTypeParser('application/json'))
+    t.assert.ifError(err)
+    t.assert.throws(() => fastify.removeContentTypeParser('application/json'))
+    fastify.close()
```

Comment:
in place of the `teardown`, we can use:

```js
t.after(() => fastify.close())
```

Reply:
done, see commit here: [c3b06e0](https://github.com/fastify/fastify/pull/5837/commits/c3b06e0d8a698a4206c80019147c64d9b42a9ac2)